### DISCUSSION TITLE
LimeSDR - Simulator - duplicate messages

### DISCUSSION THREAD

**Discussion by prefixFelix** [jopohl/urh#966] [SDR]
#### Expected Behavior
In simulation mode, several messages are to be transmitted. In this example, a delay of 5 seconds is set before the message frame to show the problem more clearly (it also occurs without a delay).

##### Simulator view:
![flow-graph](https://user-images.githubusercontent.com/62309803/142637665-3a8f6dc4-9a71-41a7-834d-ed331e0c5e5a.png)

#### Actual Behavior
If you start the simulation for the first time, everything runs as expected:
##### First start of the simulation inspected with Gqrx:
![expected](https://user-images.githubusercontent.com/62309803/142637240-52a68a3c-5204-4d46-a434-1d6101b0aa37.png)

##### First start of the simulation recorded with URH:
![expected-urh](https://user-images.githubusercontent.com/62309803/142637277-3b155bb0-f37e-47e4-84d2-e3ffe827080d.png)

If you press the start button again in the simulation window, you will get this result:
##### Second start of the simulation inspected with Gqrx:
![actual](https://user-images.githubusercontent.com/62309803/142639924-b55d712c-1788-47e7-938d-7ba3b6d52e4e.png)

##### Second start of the simulation recorded with URH:
![actual-urh](https://user-images.githubusercontent.com/62309803/142639937-1b6408ad-d9d9-4537-83c5-f13c9ef82aaf.png)

Parts the message frame (or the whole frame) are sent both shortly after the calibration phase of the LimeSDR as well as at the correct time slot. The time delay is irrelevant. If you simply remove it, more messages will be sent than are specified in the flow diagram. If you press the start button again, you will get the same result.  
Only by opening and closing the simulation window is it possible to send an valid message block again.

#### Steps To Reproduce

1. Use LimeSDR
2. Create a simple flow diagram as shown above
2. Run the simulations 2 times in a row

#### Platform Specifications

 - OS: Ubuntu 21.04 / EndeavourOS 2021.08.27
 - URH version:  2.9.2 / 2.9.2
 - Python version: 3.9.5 / 3.9.7
 - Build form source / snap

**Comment by andynoack** [comment]
I guess this is due to not flushing the TX buffer (fast enough) after sending...I am not even sure whether we could possibly fix that in URH.

**Comment by prefixFelix** [comment]
Ok, I guess it's not that big a bug, since you can just reopen the window to fix it.

**Comment by andynoack** [comment]
You are welcome to make a PR, if you think, that one is easy to do. At the moment I don't have my LimeSDR ready, so we would really appreciate any help here.
